pkg/controller/restore_controller.go

```diff
@@ -289,8 +316,19 @@ func (c *restoreController) processRestore(key string) error {
 		restore.Status.Errors += len(e)
 	}
 
+	if restoreFailure != nil {
+		logContext.Debug("restore failed")
+		restore.Status.Phase = api.RestorePhaseFailed
+		restore.Status.FailureReason = restoreFailure.Error()
+		c.metrics.RegisterRestoreFailed(backupScheduleName)
+	}
+
 	logContext.Debug("restore completed")
```

Comment:
I think L326-331 would make more sense in an `else`, since we don't want to log "restore completed" right after logging "restore failed" for failed restores, and also then you can remove L328.